### PR TITLE
Feature/improve loading process

### PR DESCRIPTION
**Pull request by JcMinarro**
### Description
Improve loading process from our offline library

### Checklist

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] Changelog updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [x] Reviewers added

### CODE REVIEW

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt [hunk 1]
```diff
@@ -435,59 +435,38 @@ internal class ChannelControllerImpl(
         val queryOfflineJob = domainImpl.scope.async { runChannelQueryOffline(pagination) }
 
         // start the online query before queryOfflineJob.await
-        val queryOnlineJob = if (domainImpl.isOnline()) {
-            domainImpl.scope.async { runChannelQueryOnline(pagination) }
-        } else {
-            // if we are not offline we mark it as needing recovery
-            recoveryNeeded = true
```
Comment (zsmb13) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt:442]: We lose this line both here and in the other controller with these changes. Do we not need to set this flag when we're offline?

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt [hunk 2]
```diff
@@ -435,59 +435,38 @@ internal class ChannelControllerImpl(
...
-        val localChannel = queryOfflineJob.await()
-        if (localChannel != null) {
+        val queryOnlineJob = domainImpl.scope.async { runChannelQueryOnline(pagination) }
+        val localChannel = queryOfflineJob.await()?.also {
             if (pagination.messageFilterDirection == Pagination.LESS_THAN) {
-                updateOldMessagesFromLocalChannel(localChannel)
+                updateOldMessagesFromLocalChannel(it)
             } else {
-                updateLiveDataFromLocalChannel(localChannel)
+                updateLiveDataFromLocalChannel(it)
```
Comment (zsmb13) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt:443]: `it` could be named better here

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt [hunk 3]
```diff
@@ -435,59 +435,38 @@ internal class ChannelControllerImpl(
...
                     recoveryNeeded = true
                 }
-            }
-            response
-        } else {
-            if (localChannel != null) {
-                Result(localChannel)
-            } else {
-                Result(ChatError("Local channel was not found"))
+                localChannel?.let { Result(it) } ?: onlineResult
```
Comment (zsmb13) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt:458]: On this branch, `onlineResult` was an error, are we sure we want to return that if we don't have local results either? (Same in the other controller)

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt [hunk 4]
```diff
@@ -435,59 +435,38 @@ internal class ChannelControllerImpl(
...
         }
         loader.value = false
         return result
     }
 
-    suspend fun runChannelQueryOffline(pagination: QueryChannelPaginationRequest): Channel? {
-        val selectedChannel = domainImpl.selectAndEnrichChannel(cid, pagination)
-
-        selectedChannel?.also { channel ->
-            channel.config = domainImpl.getChannelConfig(channel.type)
```
Comment (zsmb13) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt:484]: Do we not need this line that updates the channel config anymore?

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt (hunk 1 above)
Comment (JcMinarro) [reply stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt:442]: This flag is updated into `runChannelQueryOnline()` when the query fails, so it is not needed anymore here

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt (hunk 3 above)
Comment (JcMinarro) [reply stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt:458]: Yep, We will have a more clear error message on that case
Previously we had:
if online -> onlineResult (success/failure)
else -> if local -> localResult -> else -> Faiulre (Error without any helpful message)

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt (hunk 4 above)
Comment (JcMinarro) [reply stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt:484]: `domainImpl.selectAndEnrichChannel()` already inject `config` into the returned channel, this call was redundant

Comment (zsmb13) [reply stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt:484]: Nice catch 👍

**Review by zsmb13 — DISMISSED** [review]
LGTM

**Review by adasiewiczr — APPROVED** [review]

**Review by zsmb13 — APPROVED** [review]

**Pull request merged**